File: src/main/java/org/eln/sim/mna/process/DiodeProcess.java
package org.eln.sim.mna.process;

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln.sim.mna.component.ResistorSwitch;
import org.eln.sim.process.IProcess;

public class DiodeProcess implements IProcess {

    ResistorSwitch resistor;

    public DiodeProcess(ResistorSwitch resistor) {
        this.resistor = resistor;
    }

    @Override
    public void process(double time) {
        resistor.setState(resistor.getU() > 0);
    }
}

File: src/main/java/org/eln2/sim/mna/process/DiodeProcess.kt
package org.eln2.sim.mna.process

import org.eln2.sim.mna.component.ResistorSwitch
import org.eln2.sim.process.IProcess

open class DiodeProcess(open var resistor: ResistorSwitch) : IProcess {
    override fun process(time: Double) {
        resistor.state = resistor.u > 0
    }
}